CIScripts/Test/PesterLogger/PesterLogger.ps1

```diff
@@ -21,25 +21,57 @@ function Initialize-PesterLogger {
     $ConstOutdir = Resolve-Path $Outdir
 
     $WriteLogFunc = {
-        Param([Parameter(Mandatory = $true)] [object] $Message)
+        Param(
+            [Parameter(Mandatory = $true)] [object] $Message,
+            [parameter(ValueFromRemainingArguments=$true)] $WriterArgs,
+            [Parameter(Mandatory=$false)] [string] $Tag = "test-runner",
+            [Switch] $NoTimestamps
+        )
+
         $Scope = & $DeducerFunc
-        $Filename = ($Scope -join ".") + ".log"
+        $Filename = ($Scope -join ".") + ".txt"
         if (($Filename.IndexOfAny([System.IO.Path]::GetInvalidFileNameChars())) -ne -1) {
             throw [UnsupportedPesterTestNameException] "Invalid test name; it cannot contain some special characters, like ':', '/', etc."
         }
         $Outpath = Join-Path $Script:ConstOutdir $Filename
-        & $WriterFunc -Path $Outpath -Value $Message
+        & $WriterFunc -Path $Outpath -Value $Message -Tag $Tag -UseTimestamps (-not $NoTimestamps)
     }.GetNewClosure()
 
     Register-NewFunc -Name "Write-LogImpl" -Func $WriteLogFunc
 }
 
 function Add-ContentForce {
-    Param([string] $Path, [object] $Value)
+    Param(
+        [Parameter(Mandatory=$true)] [string] $Path,
+        [Parameter(Mandatory=$true)] [object] $Value,
+        [Parameter(Mandatory=$true)] [bool] $UseTimestamps,
+        [Parameter(Mandatory=$false)] [string] $Tag
+    )
+
     if (-not (Test-Path $Path)) {
         New-Item -Force -Path $Path -Type File | Out-Null
     }
-    Add-Content -Path $Path -Value $Value | Out-Null
+
+    $TimestampFormatString = 'yyyy-MM-dd HH:mm:ss.ffffff'
+
+    $Prefix = if ($UseTimestamps) {
+        Get-Date -Format $TimestampFormatString
+    } else {
+        " " * $TimestampFormatString.Length
+    }
+    $Prefix += " | " + $Tag + " | "
```

Comment:
I wonder if we could/would want to extract lines 55-62 to a separate function

Reply:
(64-72 maybe too?)

Reply:
Calling external functions in the closure is quite annoying, as you have to explicitly capture it. Do you think it's worth doing?

Reply:
Hmm are you sure that it needs to be done in this case? If it is required, then maybe adding a comment before each block would suffice

Reply:
Hah, I've forgotten I've already moved this logic out of the closure 😆